svthalia/concrexit
# Introduce a centralised payments page

### Summary

This PR introduces a centralised solution to be able to create payments when paying using Thalia Pay.
Since events already had such a page in place I have replaced this existing functionality so that it can be used as an example.

It basically works as follows:
A form submission is submitted to `/user/finance/payments/process/` with a body that contains
```
app_label=event
model_name=eventregistration
payable=<pk of event registration>
next=<redirection result>
```

This uses the [ContentType](https://docs.djangoproject.com/en/3.1/ref/contrib/contenttypes/#module-django.contrib.contenttypes) to dynamically obtain the model which is helped by the Payable interface we implemented before so that all these models can get a payment using this view.

### How to test
Steps to test the changes you made:
1. Create an event with registrations
2. Make sure you can pay using Thalia Pay
   - Enable the option in settings
   - Add a bank account with mandate to your user
3. Register for the event
4. Pay using Thalia Pay for your registration
